js/ui/button.js

```diff
@@ -364,6 +372,13 @@ var Button = Widget.inherit({
         }
     },
 
+    _renderStylingMode: function() {
+        var mode = this.option("stylingMode");
+        if(mode) {
+            this.$element().addClass("dx-button-mode-" + mode);
```

Comment:
We have to be sure that the value is in accepted values. If not, we have to use a default value for the current theme.